### PR TITLE
[4.1] Indexer result serializer PHP 8.1 fixes

### PR DESCRIPTION
**Pull request by laoneo**
Pull Request for pr #36482.

### Summary of Changes
This is a followup pr of #36482. 

Full credit goes to @PhilETaylor.

### Testing Instructions
Click save in the back end article form.

### Actual result BEFORE applying this Pull Request
PHP Deprecated: The Serializable interface is deprecated. Implement __serialize() and __unserialize() instead (or in addition, if support for old PHP versions is necessary)

### Expected result AFTER applying this Pull Request
No error.

### CODE REVIEW

File: administrator/components/com_finder/src/Indexer/Result.php [hunk 1]
```diff
@@ -508,42 +532,43 @@ public function serialize()
...
+			'published',
+			'route',
+			'sale_price',
+			'start_date',
+			'state',
+			'taxonomy',
+			'title',
+			'type_id',
+			'url',
+			];
```
Comment (Quy) [inline administrator/components/com_finder/src/Indexer/Result.php]:
```suggestion
		];
```

**Comment by MacJoom** [comment]
I have tested this item :white_check_mark: successfully on ca81ba410494830a0840b68a70725c168c464662<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37338">issues.joomla.org/tracker/joomla-cms/37338</a>.</sub>

**Comment by RickR2H** [comment]
I have tested this item :white_check_mark: successfully on ca81ba410494830a0840b68a70725c168c464662<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37338">issues.joomla.org/tracker/joomla-cms/37338</a>.</sub>

**Comment by RickR2H** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37338">issues.joomla.org/tracker/joomla-cms/37338</a>.</sub>

**Comment by bembelimen** [comment]
Thx

**Pull request merged**